test/unit/workDiv/src/WorkDivForKernelTest.cpp

```diff
@@ -0,0 +1,224 @@
+    auto const dev = alpaka::getDevByIdx(platform, 0);
+
+    TestKernelWithManyRegisters kernel;
+    size_t val(200ul);
+    auto const& bundeledKernel = alpaka::KernelBundle(kernel, val);
+
+    // Get hard limits for test
+    auto const& props = alpaka::getAccDevProps<Acc>(dev);
+    Idx const threadsPerGridTestValue = props.m_blockThreadCountMax * props.m_gridBlockCountMax;
+
+    // Test getValidWorkDivForKernel function for threadsPerGridTestValue threads per grid.
+    auto const workDiv
+        = alpaka::getValidWorkDivForKernel<Acc>(dev, bundeledKernel, Vec{8, threadsPerGridTestValue / 8}, Vec{1, 1});
+
+    // Test isValidWorkDivKernel function
+    auto const isValid = alpaka::isValidWorkDivKernel<Acc>(dev, bundeledKernel, workDiv);
+    CHECK(isValid == true);
+
+    if constexpr(alpaka::accMatchesTags<Acc, alpaka::TagGpuCudaRt>)
+    {
+        // Expected valid workdiv for this kernel. These values might change depending on the GPU type and compiler
+        // therefore commented out. Number of registers used by kernel might limit threads per block value differently
+        // for different GPUs. CHECK(workDiv == WorkDiv{Vec{8, 729444}, Vec{1, 736}, Vec{1, 1}});
+
+        // Get calculated threads per block from the workDiv found by examining kernel function
+        auto const threadsPerBlock = workDiv.m_blockThreadExtent.prod();
+        // Get hard limits
+        auto const threadsPerBlockLimit = props.m_blockThreadCountMax;
+
+        // Depending on the GPU type or the compiler the test below might fail because threadsPerBlock can be equal to
+        // threadsPerBlockLimit, which is the max device limit.
+        CHECK(threadsPerBlock < static_cast<Idx>(threadsPerBlockLimit));
+
+        // too many threads per block
+        auto const invalidWorkDiv
+            = WorkDiv{Vec{8, threadsPerGridTestValue / 8}, Vec{2 * threadsPerBlock, 1}, Vec{1, 1}};
+        auto isWorkDivValidForCuda = alpaka::isValidWorkDivKernel<Acc>(dev, bundeledKernel, invalidWorkDiv);
+        CHECK(isWorkDivValidForCuda == false);
+
+        auto const validWorkDiv = WorkDiv{Vec{8, threadsPerGridTestValue / 8}, Vec{1, threadsPerBlock}, Vec{1, 1}};
+        isWorkDivValidForCuda = alpaka::isValidWorkDivKernel<Acc>(dev, bundeledKernel, validWorkDiv);
+        CHECK(isWorkDivValidForCuda == true);
+    }
+    else if constexpr(alpaka::accMatchesTags<Acc, alpaka::TagGpuHipRt>)
+    {
+        // Get calculated threads per block from the workDiv found by examining the kernel function
+        auto const threadsPerBlock = workDiv.m_blockThreadExtent.prod();
+        // Get hard limits
+        auto const threadsPerBlockLimit = props.m_blockThreadCountMax;
+        // Depending on the GPU type or the compiler this test might fail because threadsPerBlock can be less than
+        // threadsPerBlockLimit, which is the max device limit.
+        CHECK(threadsPerBlock == static_cast<Idx>(threadsPerBlockLimit));
```

Comment:
CI is saying this is failing, IMO you should do the same you did for CUDA. So you can most likely merge the CUDA and HIP if branch.

```
: /builds/hzdr/crp/alpaka/test/unit/workDiv/src/WorkDivForKernelTest.cpp:198: FAILED:
48:   CHECK( threadsPerBlock == static_cast<Idx>(threadsPerBlockLimit) )
48: with expansion:
48:   736 (0x2e0) == 1024 (0x400)
```